components/camera/collectors_test.go

```diff
@@ -0,0 +1,208 @@
+
+const (
+	serviceName     = "camera"
+	captureInterval = time.Millisecond
+)
+
+func convertStringMapToAnyPBMap(params map[string]string) (map[string]*anypb.Any, error) {
+	methodParams := map[string]*anypb.Any{}
+	for key, paramVal := range params {
+		anyVal, err := convertStringToAnyPB(paramVal)
+		if err != nil {
+			return nil, err
+		}
+		methodParams[key] = anyVal
+	}
+	return methodParams, nil
+}
+
+func convertStringToAnyPB(str string) (*anypb.Any, error) {
+	var wrappedVal protoreflect.ProtoMessage
+	if boolVal, err := strconv.ParseBool(str); err == nil {
+		wrappedVal = wrapperspb.Bool(boolVal)
+	} else if int64Val, err := strconv.ParseInt(str, 10, 64); err == nil {
+		wrappedVal = wrapperspb.Int64(int64Val)
+	} else if uint64Val, err := strconv.ParseUint(str, 10, 64); err == nil {
+		wrappedVal = wrapperspb.UInt64(uint64Val)
+	} else if float64Val, err := strconv.ParseFloat(str, 64); err == nil {
+		wrappedVal = wrapperspb.Double(float64Val)
+	} else {
+		wrappedVal = wrapperspb.String(str)
+	}
+	anyVal, err := anypb.New(wrappedVal)
+	if err != nil {
+		return nil, err
+	}
+	return anyVal, nil
+}
+
+func TestCollectors(t *testing.T) {
+	logger := logging.NewTestLogger(t)
+	methodParams, err := convertStringMapToAnyPBMap(map[string]string{"camera_name": "camera-1", "mime_type": "image/jpeg"})
+	test.That(t, err, test.ShouldBeNil)
+	viamLogoJpeg, err := io.ReadAll(base64.NewDecoder(base64.StdEncoding, bytes.NewReader(viamLogoJpegB64)))
+	test.That(t, err, test.ShouldBeNil)
+	viamLogoJpegAsInts := []any{}
+	for _, b := range viamLogoJpeg {
+		viamLogoJpegAsInts = append(viamLogoJpegAsInts, int(b))
+	}
+
+	img := rimage.NewLazyEncodedImage(viamLogoJpeg, utils.MimeTypeJPEG)
+	// 32 x 32 image
+	test.That(t, img.Bounds().Dx(), test.ShouldEqual, 32)
+	test.That(t, img.Bounds().Dy(), test.ShouldEqual, 32)
+
+	pcd, err := pointcloud.NewFromFile(artifact.MustPath("pointcloud/test.las"), logger)
+	test.That(t, err, test.ShouldBeNil)
+
+	var pcdBuf bytes.Buffer
+	test.That(t, pointcloud.ToPCD(pcd, &pcdBuf, pointcloud.PCDBinary), test.ShouldBeNil)
+
+	now := time.Now()
+	nowPB := timestamppb.New(now)
+	cam := newCamera(img, img, now, pcd)
```

Comment:
you always test on a camera where `colorImage`==`depthImage`, how about we have 2 different images for completeness? (if it's easier, one can be the reversed bytes of the other 🤷)

Reply:
ack! reversing the bytes was in re my suggestion to use just arbitrary bytes as images, but this issue makes sense as you've explained we assume an `image.Image` in the camera SDK.